src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/UseInCSharpContextOnly.cs

```diff
@@ -0,0 +1,42 @@
+namespace DotNetProjectFile.Analyzers.MsBuild;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class UseInCSharpContextOnly() : MsBuildProjectFileAnalyzer(Rule.UseInCSharpContextOnly)
+{
+    private static readonly Type[] CSharpOnly =
+    [
+        typeof(DotNetProjectFile.MsBuild.CSharp.AllowUnsafeBlocks),
+        typeof(DotNetProjectFile.MsBuild.CSharp.CheckForOverflowUnderflow),
+        typeof(DotNetProjectFile.MsBuild.CSharp.Nullable),
+    ];
+
+    protected override IReadOnlyCollection<ProjectFileType> ApplicableTo => ProjectFileTypes.ProjectFile_DirectoryBuild;
+
+    protected override void Register(ProjectFileAnalysisContext<MsBuildProject> context)
+    {
+        if (!InCSharpContext(context.File))
+        {
+            Walk(context.File, context);
+        }
+    }
+
+    private static bool InCSharpContext(MsBuildProject project) => project.FileType switch
+    {
+        ProjectFileType.ProjectFile => project.Path.Extension.IsMatch(".csproj"),
+        ProjectFileType.DirectoryBuild => !project.Path.Directory.Files("*.vbproj").Any(),
```

Comment:
In the vbonly rule you check if there are any vbproj files, shouldn't you check here if there are any csproj files

Reply:
No. There are more extension types which can be other languages, but also other C# based types, where there is (as far as I know) only one type of project file extension linked to VB.NET. So hence the asymmetrical implementation.